google/cloud/security/inventory/inventory_loader.py

```diff
@@ -372,45 +242,67 @@ def _send_email(cycle_time, cycle_timestamp, status, pipelines,
         LOGGER.error('Unable to send email that inventory snapshot completed.')
 
 
-def _configure_logging(configs):
+def _configure_logging(flag_configs):
     """Configures the loglevel for all loggers."""
-    desc = configs.get('loglevel')
+    desc = flag_configs.get('loglevel')
     level = LOGLEVELS.setdefault(desc, 'info')
     log_util.set_logger_level(level)
 
 def main(_):
     """Runs the Inventory Loader."""
+
+    flag_configs = FLAGS.FlagValuesDict()
+    _configure_logging(flag_configs)
+
+    if FLAGS.pipeline_configs is None:
+        LOGGER.error('Path to pipeline config needs to be specified.')
+        sys.exit()
+
+    # Make these re-usable so that the rate-limiter can apply across
+    # different pipelines.
     try:
-        dao = Dao()
-        project_dao = proj_dao.ProjectDao()
-        organization_dao = org_dao.OrganizationDao()
-        bucket_dao = buck_dao.BucketDao()
-        cloudsql_dao = sql_dao.CloudsqlDao()
-        fwd_rules_dao = fr_dao.ForwardingRulesDao()
-        folder_dao = folder_resource_dao.FolderDao()
+        api_map = {
+            'admin_api': admin_directory.AdminDirectoryClient(),
+            'bigquery_api': bigquery.BigQueryClient(),
+            'cloudsql_api': cloudsql.CloudsqlClient(),
+            'compute_api': compute.ComputeClient(),
+            'compute_beta_api': compute.ComputeClient(version='beta'),
+            'crm_api': crm.CloudResourceManagerClient(),
+            'crm_v2beta1': crm.CloudResourceManagerClient(version='v2beta1'),
+            'gcs_api': gcs.StorageClient(),
```

Comment:
Does it make sense to eventually (not necessarily for this PR) make inventory loader not instantiate daos and clients before running the pipelines, and move the dao/api client instantiation to the pipelines? As inventory loader becomes more of a pure "loader"/"runner" module, it seems like it should not know about the internals of the pipelines and just pass configuration details to each pipeline.